### PR TITLE
N°6239 - Portail: Fix error message when trying to display object with missing icon

### PR DESCRIPTION
**Pull request by accognet**
internal
Error message: 
image

### CODE REVIEW

File: core/attributedef.class.inc.php [hunk 1]
```diff
@@ -8402,8 +8402,11 @@ public function MakeFormField(DBObject $oObject, $oFormField = null)
 		}
 		else
 		{
-			$oFormField->SetDownloadUrl($this->Get('default_image'));
-			$oFormField->SetDisplayUrl($this->Get('default_image'));
+			$sDefaultImg = $this->Get('default_image');
+			if (is_object($sDefaultImg) && !$sDefaultImg->IsEmpty()) {
+				$oFormField->SetDownloadUrl($sDefaultImg);
+				$oFormField->SetDisplayUrl($sDefaultImg);
```
Comment (Molkobain) [inline core/attributedef.class.inc.php]: `$oDefaultImage`

File: core/attributedef.class.inc.php [hunk 2]
```diff
@@ -8402,8 +8402,11 @@ public function MakeFormField(DBObject $oObject, $oFormField = null)
 		}
 		else
 		{
-			$oFormField->SetDownloadUrl($this->Get('default_image'));
-			$oFormField->SetDisplayUrl($this->Get('default_image'));
+			$oDefaultImg = $this->Get('default_image');
+			if (is_object($oDefaultImg) && !$oDefaultImg->IsEmpty()) {
+				$oFormField->SetDownloadUrl($oDefaultImg);
+				$oFormField->SetDisplayUrl($oDefaultImg);
+			}
```
Comment (Molkobain) [inline core/attributedef.class.inc.php]:
```suggestion
			$oDefaultImage = $this->Get('default_image');
			if (is_object($oDefaultImage) && !$oDefaultImage->IsEmpty()) {
				$oFormField->SetDownloadUrl($oDefaultImage);
				$oFormField->SetDisplayUrl($oDefaultImage);
			}
```

**Comment by Molkobain** [comment]
Can you post a screenshot of how it renders when there is no default image?

**Comment by accognet** [comment]
There is only an error message in the screen and the object can not be displayed.

**Comment by piRGoif** [comment]
Technical review : impl OK !
In the default datamodel there are 5 AttributeImage, only the one in Person has a default_image. Bug might be easy to reproduce...  Before functional review next week, please write down a precise step by step guide to reproduce on a default iTop cty instance.

**Comment by Molkobain** [comment]
Accepted during functional review for 3.1.0

**Pull request merged**